testing/python/collect.py

```diff
@@ -1135,21 +1157,21 @@ def test_foo(self):
                     pass
         """
         )
-        classcol = testdir.collect_by_name(modcol, "TestClass")
-        instance = classcol.collect()[0]
+        classcol = pytester.collect_by_name(modcol, "TestClass")
+        instance = list(classcol.collect())[0]  # type: ignore[union-attr]
```

Comment:
Can we add some assert here instead of the type-ignore?

Reply:
added assertions